File: dev-docs/snippets/workflows/start.swift
import JsBaoClient

// Start a workflow run. On Swift `input` is a positional `[String: Any]`
// and the idempotency/scoping fields live on `StartWorkflowOptions`.
func start(client: JsBaoClient, workflowKey: String, contextDocId: String) async throws {
  // #region example
  let run = try await client.workflows.start(
    workflowKey: workflowKey,
    input: ["topic": "spring planting"],
    options: StartWorkflowOptions(runKey: "run-001", contextDocId: contextDocId)
  )
  // run["runId"] / run["runKey"] / run["status"] / run["existing"]
  // #endregion example
  _ = run
}

File: dev-docs/snippets/workflows/start.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Start a workflow run. `runKey` makes the start idempotent.
export async function start(
  client: JsBaoClient,
  workflowKey: string,
  contextDocId: string,
) {
  // #region example
  const run = await client.workflows.start({
    workflowKey,
    input: { topic: "spring planting" },
    runKey: "run-001",
    contextDocId,
  });
  // run.runId / run.runKey / run.status / run.existing
  // #endregion example
  return run;
}
